gnomad/utils/transcript_annotation.py

```diff
@@ -53,48 +52,122 @@ def summarize_transcript_expression(
+    :param ht: Table of summarized transcript expression by tissue.
+    :return: Table with expression proportion of transcript to gene per tissue
+        and mean expression proportion across tissues.
+    """
+    tissues = list(ht.row_value)
 
-    gene_ht = transcript_ht.group_by("gene_id").aggregate(
+    # Calculate the sum of transcript expression by gene per tissue.
+    gene_ht = ht.group_by("gene_id").aggregate(
+        **{tissue: hl.agg.sum(ht[tissue]) for tissue in tissues}
+    )
+
+    # Return the proportion of expression of transcript to gene per tissue.
+    gene = gene_ht[ht.gene_id]
+    return hl.struct(
         **{
-            tissue: hl.agg.sum(transcript_ht[tissue])
-            for tissue in list(transcript_ht.row_value)
+            tissue: hl.utils.misc.divide_null(ht[tissue], gene[tissue])
+            for tissue in tissues
         }
     )
 
-    return transcript_ht, gene_ht
+
+def filter_expression_ht_by_tissues(
+    ht: hl.Table,
+    tissues_to_keep: Optional[List[str]] = None,
+    tissues_to_filter: Optional[List[str]] = None,
+) -> hl.Table:
+    """
+    Filter a Table with a row annotation for each tissue to only include specified tissues.
+
+    :param ht: Table with a row annotation for each tissue.
+    :param tissues_to_keep: Optional list of tissues to keep in the Table. Default is
+        all non-key rows in the Table.
+    :param tissues_to_filter: Optional list of tissues to exclude from the Table.
+    :return: Table with only specified tissues.
+    """
+    if tissues_to_keep is None and tissues_to_filter is None:
+        logger.info(
+            "No tissues_to_keep or tissues_to_filter specified. Returning input Table."
+        )
+        return ht
+
+    if tissues_to_keep is None:
+        tissues = list(ht.row_value)
+
+    if tissues_to_filter is not None:
+        logger.info("Filtering tissues: %s", tissues_to_filter)
+        tissues = [t for t in tissues if t not in tissues_to_filter]
+
+    ht = ht.select(*tissues)
+
+    return ht
 
 
 def tissue_expression_ht_to_array(
     ht: hl.Table,
-    tissues: Optional[List[str]] = None,
+    tissues_to_keep: Optional[List[str]] = None,
     tissues_to_filter: Optional[List[str]] = None,
+    annotations_to_extract: Optional[Union[Tuple[str], List[str]]] = (
+        "transcript_expression",
+        "expression_proportion",
+    ),
 ) -> hl.Table:
     """
     Convert a Table with a row annotation for each tissue to a Table with tissues as an array.
 
-    The output is a Table with a field 'tissue_expression' containing an array of
-    summarized expression values by tissue, where the order of tissues in the array is
-    indicated by the "tissues" global annotation.
+    The output is a Table with fields in `annotations_to_extract`,
+    each containing an array of summarized expression values or proportion
+    by tissue, where the order of tissues in the array is indicated by
+    the "tissues" global annotation.
```

Comment:
Not exactly, it either has an annotation of `tissue_expression` that is an array of information per tissue, where each element of the array is the Table's row value for a given tissue. 

Or it can be a Table with one array annotation for each field defined in `annotations_to_extract`, where each array is an array of the given field value per Tissue. 

Maybe an example of the two options would make it clear?